x/interchaintxs/keeper/msg_server.go

```diff
@@ -83,6 +83,12 @@ func (k Keeper) SubmitTx(goCtx context.Context, msg *ictxtypes.MsgSubmitTx) (*ic
 		return nil, sdkerrors.Wrapf(ictxtypes.ErrNotContract, "%s is not a contract address", msg.FromAddress)
 	}
 
+	feePayerAddr, err := sdk.AccAddressFromBech32(msg.Fee.Payer)
+	if msg.Fee.Payer != "" && err != nil {
+		k.Logger(ctx).Debug("SubmitTx: failed to parse fee payer address", "fee payer", msg.Fee.Payer)
+		return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "failed to parse address: %s", msg.Fee.Payer)
+	}
+
```

Comment:
Just a suggestion: maybe we can define a method which accepts `sender` and `payer` and returns `(feetypes.PayerInfo, error)`?

Reply:
+1, cause it looks a bit hard to read